cfn-resources/search-index/cmd/resource/resource.go

```diff
@@ -205,18 +206,31 @@ func newMappingsFields(fields []string) (map[string]interface{}, error) {
 
 	fieldsMap := make(map[string]interface{})
 	for _, p := range fields {
-		f := strings.Split(p, ":")
+		f := strings.SplitN(p, ":", 2)
 		if len(f) != indexFieldParts {
 			return nil, fmt.Errorf("partition should be fieldName:fieldType, got: %s", p)
 		}
-		fieldsMap[f[0]] = map[string]interface{}{
-			"type": f[1],
+
+		fieldType, err := parseFieldType(f[1])
+		if err != nil {
+			return nil, err
 		}
+
+		fieldsMap[f[0]] = fieldType
```

Comment:
curious: what has been the testing strategy here? Is the only available test the one you did manually in the PR description?

Reply:
ive added a contract testing for this one